File: BE-Task_1-Java/src/main/java/io/mekari/assessment/refundbackendjava/repository/PurchaseRepository.java
package io.mekari.assessment.refundbackendjava.repository;

import io.mekari.assessment.refundbackendjava.domain.model.Purchase;
import java.util.List;
import java.util.Optional;

public interface PurchaseRepository {
    Optional<Purchase> findById(String purchaseId);

    List<Purchase> listAll();
}

File: BE-Task_1-Php/internal/repository/interfaces/purchase_repository.php
<?php

declare(strict_types=1);

namespace App\Internal\Repository\Interfaces;

use App\Internal\Domain\Entities\Purchase;

interface PurchaseRepositoryInterface
{
    public function findById(string $purchaseId): ?Purchase;

    /**
     * @return list<Purchase>
     */
    public function listAll(): array;
}
